lookup_plugin/helper.py

```diff
@@ -0,0 +1,141 @@
+    base_dir = cluster_params['base_dir']
+    raft_uuid = cluster_params['raft_uuid']
+    baseDir = os.path.join(base_dir, raft_uuid)
+
+    if seed is not None:
+        dbi_dir = os.path.join(baseDir, dirName, seed)
+    else:
+        dbi_dir = os.path.join(baseDir, dirName)
+
+    # Get a list of all entries (files and directories) under the 'dbi-dbo' directory
+    entries = os.listdir(dbi_dir)
+
+    # Filter out directories only
+    directories = [entry for entry in entries if os.path.isdir(os.path.join(dbi_dir, entry))]
+
+    if directories:
+        # Sort directories based on creation time (most recent first)
+        directories.sort(key=lambda d: os.path.getctime(os.path.join(dbi_dir, d)), reverse=True)
+        # Return the path of the most recently created directory with a trailing slash
+        most_recent_directory = directories[0]
+        if len(directories) > 1:
+            most_recent_directory = directories[1]
+        directory_path = os.path.join(dbi_dir, most_recent_directory, '')  # Add the trailing slash here
+        return directory_path
+    else:
+        return None
+
+def list_files_from_dir(dir_path):
+    return os.listdir(dir_path)    
+
+def copy_DBI_file_generatorNum(cluster_params, dirName, chunk):
+    jsonPath = get_dir_path(cluster_params, dirName)
+    jsonfile = get_dummy_gen_config_path(jsonPath, chunk)
+    json_data = load_parameters_from_json(jsonfile)
+    dbi_input_path = str(json_data['DbiPath'])
+    # List files from dbipath
+    files_list = list_files_from_dir(dbi_input_path)
+    
+    if files_list:
+        # Select a random file from the list
+        random_file = random.choice(files_list)
+        filename_parts = random_file.split(".")
+        # Extract the generation number
+        genration_num = filename_parts[GEN_NUM]
+        # Increment the extracted element by 1
+        # Decrement as the number is inversed
+        new_genration_num = str(int(genration_num, 16) - 1)
+        # Update the filename with the incremented element
+        filename_parts[GEN_NUM] = new_genration_num
+        new_filename = ".".join(filename_parts)
+        source_file_path = os.path.join(dbi_input_path, random_file)
+        new_file_path = os.path.join(dbi_input_path, new_filename)
+        print(f"Copying {source_file_path} to {new_file_path}") 
+        # Copy the file and rename the copy
+        copy_files(source_file_path, new_file_path)
+    else:
+        print("No files found in the directory.")
+     
+
+
+class LookupModule(LookupBase):
+    def run(self, terms, **kwargs):
+        operation = terms[0]
+        cluster_params = kwargs['variables']['ClusterParams']
+        
+        if operation == "clone_dbi_set":
+            chunk = terms[1]
+            clone_dbi_files(cluster_params, chunk)
+        if operation == "copy_DBI_file_generatorNum":
+            chunk = terms[1]
+            copy_DBI_file_generatorNum(cluster_params, DBI_DIR, chunk)
```

Comment:
follow the snake case pattern for function naming.
DBI_DIR is a global variable. so we don't need to pass it as a function parameter